### PR TITLE
Incorporate intel-tbb-oneapi package into intel-tbb package

### PR DESCRIPTION
**Pull request by gartung**
The intel-tbb-oneapi package uses cmake to configure the build for tbb v2021.1.1+
This package is incorporated into the intel-tbb package using `@when` decorators.

### CODE REVIEW

**Comment by adamjstewart** [comment]
So `intel-oneapi-tbb` and `intel-tbb-oneapi` are the same software? I would make them the same package. We generally want to compile from source whenever possible, so I think it's fine to overwrite the original package with your build recipe.

File: var/spack/repos/builtin/packages/intel-oneapi-tbb/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/intel-oneapi-tbb/package.py]: Let's add @rscohn2 back as a maintainer

**Comment by tgamblin** [comment]
@rscohn2 any objections to making this PR into a refactor of `intel-oneapi-tbb` instead of a separate package?

File: var/spack/repos/builtin/packages/intel-oneapi-tbb/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/intel-oneapi-tbb/package.py]: This shouldn't be necessary, Spack can correctly parse the url:
```console
$ spack url parse https://github.com/oneapi-src/oneTBB/archive/v2020.1.tar.gz
==> Parsing URL: https://github.com/oneapi-src/oneTBB/archive/v2020.1.tar.gz

==> Matched version regex  1: r'^v?(\\d[\\d._-]*)$'
==> Matched  name   regex  0: r'github\\.com/[^/]+/([^/]+)'

==> Detected:
    https://github.com/oneapi-src/oneTBB/archive/v2020.1.tar.gz
                                  ------          ~~~~~~
    name:    oneTBB
    version: 2020.1

==> Substituting version 9.9.9b:
    https://github.com/oneapi-src/oneTBB/archive/v9.9.9b.tar.gz
                                  ------          ~~~~~~
```

File: var/spack/repos/builtin/packages/intel-oneapi-tbb/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/intel-oneapi-tbb/package.py]: We can just have a single `url` now that `url_prefix` isn't used anywhere else

**Review by adamjstewart — DISMISSED** [review]
This looks good to me now, but I'll wait for @rscohn2 to review before merging since this drastically changes the package.

**Comment by rscohn2** [comment]
Product builds and open source builds of oneapi are not interchangeable. They should be separate packages. In fact, there already is a spack package that builds tbb from open source: https://spack.readthedocs.io/en/latest/package_list.html#intel-tbb It is out of date and would benefit from the work that @gartung did.

**Comment by rscohn2** [comment]
Both intel-tbb and intel-oneapi-tbb do a provides('tbb') so they are interchangeable in that sense.

**Comment by tgamblin** [comment]
> Product builds and open source builds of oneapi are not interchangeable. 

Can you elaborate on this one? Are the product builds proprietary (i.e., they have things that are not in the OSS versions, that a user could not build) or is this just to say that the product builds are the ones Intel vouches for?

**Comment by rscohn2** [comment]
> Are the product builds proprietary (i.e., they have things that are not in the OSS versions, that a user could not build) or is this just to say that the product builds are the ones Intel vouches for?

Both are true. Many times, products are built from open source without modification, but it can change from release to release and product by product. Separate packages with a common virtual package seems like the best way to let the users choose.

**Comment by tgamblin** [comment]
Ok I agree with @rscohn2 then -- lets' keep the `intel-oneapi` prefix as the official product and modify `intel-tbb` with these updates.  Work for you @gartung and @adamjstewart?

**Comment by gartung** [comment]
The intel-tbb package for releases before v2020.1 uses cmake to configure the build rather then environment variables and make. I would prefer to keep this as a separate package. Maybe I should name it intel-tbb-cmake?
@tgamblin @adamjstewart any opinions?

**Comment by adamjstewart** [comment]
See https://spack.readthedocs.io/en/latest/build_systems/multiplepackage.html for our official docs on supporting multiple build systems! One of the options is to append the build system to the package name, but we specifically explain why that's a bad idea and offer a better suggestion instead.

I'm excited that I finally get to use these docs!

**Comment by gartung** [comment]
Verified that I could build v2020.3 and v2021.1.1 on macOS and Centos8.

**Comment by gartung** [comment]
@adamjstewart  or @tgamblin can you review please

**Review by tgamblin — APPROVED** [review]

**Pull request merged**
